File: psp/psp1Eva/PSP1Eva/cliente/src/main/java/cliente/ui/pantallas/principal/PrincipalController.java
package cliente.ui.pantallas.principal;

import cliente.ui.common.BasePantallaController;
import cliente.ui.common.Pantallas;
import jakarta.enterprise.inject.Instance;
import jakarta.inject.Inject;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.fxml.Initializable;
import javafx.scene.control.Alert;
import javafx.scene.control.Menu;
import javafx.scene.control.MenuItem;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.Pane;
import lombok.extern.log4j.Log4j2;

import java.io.IOException;
import java.net.URL;
import java.util.ResourceBundle;

@Log4j2
public class PrincipalController implements Initializable {

    @FXML
    private MenuItem itemGames;
    @FXML
    private MenuItem itemShops;
    @FXML
    private Menu goTo;
    @FXML
    public BorderPane root;
    Instance<Object> instance;
    private final Alert alert;

    @Inject
    public PrincipalController(Instance<Object> instance) {
        this.instance = instance;
        alert = new Alert(Alert.AlertType.NONE);
    }

    private Pane cargarPantalla(String ruta) {
        Pane panePantalla = null;
        try {

            FXMLLoader fxmlLoader = new FXMLLoader();
            fxmlLoader.setControllerFactory(controller -> instance.select(controller).get());
            panePantalla = fxmlLoader.load(getClass().getResourceAsStream(ruta));
            BasePantallaController pantallaController = fxmlLoader.getController();
            pantallaController.setPrincipalController(this);
            pantallaController.principalCargado();


        } catch (IOException e) {
            log.error(e.getMessage(), e);
        }
        return panePantalla;
    }

    @Override
    public void initialize(URL url, ResourceBundle resourceBundle) {
        goShops();
    }

    private void cambioPantalla(Pane pantallaNueva) {
        root.setCenter(pantallaNueva);
    }

    private void cargarPantalla(Pantallas pantalla) {
        cambioPantalla(cargarPantalla(pantalla.getRuta()));
    }

    public void goShops() {
        cargarPantalla(Pantallas.SHOPS);
    }

    public void goGames() {
        cargarPantalla(Pantallas.GAMES);
    }

    public void error(String mensaje) {
        alert.setAlertType(Alert.AlertType.ERROR);
        alert.setContentText(mensaje);
        alert.showAndWait();
    }
}

File: psp/basicGraphql/graphqlClient/graphqlClient/src/main/kotlin/org/example/ui/controllers/PrincipalController.kt
package org.example.ui.controllers

import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.scene.control.Alert
import javafx.scene.layout.BorderPane
import javafx.scene.layout.Pane
import javafx.stage.Stage
import javafx.stage.WindowEvent
import lombok.extern.log4j.Log4j2
import org.example.ui.common.BaseScreenController
import org.example.ui.common.Screens
import java.io.IOException
import kotlin.system.exitProcess

@Log4j2
class PrincipalController {

    @FXML
    var root: BorderPane? = null

    private var alert: Alert = Alert(Alert.AlertType.NONE)

    private var primaryStage: Stage? = null

    fun showAlertError(mensaje: String?) {
        alert.alertType = Alert.AlertType.ERROR
        alert.contentText = mensaje
        alert.isResizable = true
        alert.showAndWait()
    }

    fun showAlertInformation(message: String?) {
        val infoAlert = Alert(Alert.AlertType.INFORMATION)
        infoAlert.contentText = message
        infoAlert.showAndWait()
    }

    fun showLoading(show: Boolean = true) {
        if (show) {
            root!!.cursor = javafx.scene.Cursor.WAIT
        } else {
            root!!.cursor = javafx.scene.Cursor.DEFAULT
        }
    }

    private fun cargarPantalla(ruta: String): Pane? {
        var panePantalla: Pane? = null
        try {
            val fxmlLoader = FXMLLoader()
            panePantalla = fxmlLoader.load(javaClass.getResourceAsStream(ruta))
            val pantallaController: BaseScreenController = fxmlLoader.getController()
            pantallaController.setPrincipalController(this)
            pantallaController.principalCargado()
        } catch (e: IOException) {
            showAlertError("Error al cargar la pantalla: $ruta ($e)")
        }
        cambioPantalla(panePantalla)
        return panePantalla
    }

    fun setStage(stage: Stage?) {
        primaryStage = stage
        primaryStage!!.addEventFilter(
            WindowEvent.WINDOW_CLOSE_REQUEST
        ) {
            primaryStage!!.close()
            exitProcess(0)
        }
    }

    private fun cambioPantalla(pantallaNueva: Pane?) {
        root!!.center = pantallaNueva
    }

    fun initialize() {
        goShops()
    }

    private fun cargarPantalla(pantalla: Screens) {
        cambioPantalla(cargarPantalla(pantalla.ruta))
    }

    fun goShops() {
        cargarPantalla(Screens.SHOPS)
    }

    fun goGames() {
        cargarPantalla(Screens.GAMES)
    }

}